cpp/src/copying/shift.cu

```diff
@@ -116,37 +116,22 @@ struct functor {
 
 } // anonymous namespace
 
-std::unique_ptr<table> shift(table_view const& input,
-                             size_type offset,
-                             std::vector<std::reference_wrapper<scalar>> const& fill_values,
-                             rmm::mr::device_memory_resource *mr,
-                             cudaStream_t stream)
+std::unique_ptr<column> shift(column_view const& input,
+                              size_type offset,
+                              scalar const& fill_value,
+                              rmm::mr::device_memory_resource *mr,
+                              cudaStream_t stream)
 {
-    CUDF_EXPECTS(input.num_columns() == static_cast<size_type>(fill_values.size()),
-                 "shift requires one fill value for each column.");
+    CUDF_EXPECTS(input.type() == fill_value.type(),
+                "shift requires each fill value type to match the corrosponding column type.");
 
-    for (size_type i = 0; i < input.num_columns(); ++i) {
-        CUDF_EXPECTS(input.column(i).type() == fill_values[i].get().type(),
-                     "shift requires each fill value type to match the corrosponding column type.");
-    }
-
-    if (input.num_rows() == 0) {
+    if (input.size() == 0) {
         return empty_like(input);
     }
 
-    auto output_columns = std::vector<std::unique_ptr<column>>{};
-
-    std::transform(input.begin(),
-                   input.end(),
-                   fill_values.begin(),
-                   std::back_inserter(output_columns),
-                   [offset, mr, stream] (auto const& input_column, auto const& fill_value) {
-                       return type_dispatcher(input_column.type(), functor{},
-                                              input_column, offset, fill_value,
-                                              mr, stream);
-                   });
-
-    return std::make_unique<table>(std::move(output_columns));
+    return type_dispatcher(input.type(), functor{},
```

Comment:
Not your change, but "functor" is a pretty uninformative name.

Reply:
Technically my change from a previous PR. Contextually, and with the established per-top-level-function dispatch pattern, I thought it made sense, but changing it is easy.